hedwig/backends/gcp.py

```diff
@@ -196,6 +196,16 @@ def shutdown(self, await_msg_callbacks=False) -> None:
         # instead, it's faster to actually process all the messages
 
 
+class ShutdownMessageScheduler(PubSubMessageScheduler):
+    def __init__(self, work_queue: Queue, subscription_path: str, shutdown_event: threading.Event):
+        super().__init__(work_queue, subscription_path)
+        self.shutdown_event = shutdown_event
+
+    def shutdown(self, await_msg_callbacks=False) -> List:  # type: ignore
+        self.shutdown_event.set()
```

Comment:
Not sure if setting this shutdown event in the Scheduler is necessary. I think the important bit was to return the empty list instead of `None`.